activation/poet.go

```diff
@@ -134,6 +142,14 @@ func NewHTTPPoetClient(server types.PoetServer, cfg PoetConfig, opts ...PoetClie
 		CheckRetry:   checkRetry,
 	}
 
+	submitChallengeClient := &retryablehttp.Client{
+		RetryMax:     math.MaxInt,
+		RetryWaitMin: cfg.RequestRetryDelay,
+		RetryWaitMax: 2 * cfg.RequestRetryDelay,
+		Backoff:      retryablehttp.LinearJitterBackoff,
```

Comment:
NIT: I'm not 100% sure if the linear jitter backoff will work well here because the retry delay grows without bounds. In extreme cases (with many retries) the delay can become long. Perhaps use the default one: https://pkg.go.dev/github.com/hashicorp/go-retryablehttp@v0.7.7#DefaultBackoff? It limits the delay to the `RetryWaitMax` (then, `RetryWaitMax` needs to be adjusted too).